data-collection/data-collection/View Controllers/Map View Controller/MapViewController+Extras.swift

```diff
@@ -0,0 +1,126 @@
+//// Copyright 2020 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import ArcGIS
+import ArcGISToolkit
+
+private let layersExtra = "Layers"
+private let bookmarksExtra = "Bookmarks"
+
+/// Defines the Extras.
+public enum Extras : CaseIterable {
+    // Displays layer content.
+    case layers
+    // Displays bookmarks.
+    case bookmarks
+}
+
+extension Extras : CustomStringConvertible {
+    public var description: String {
+        switch self {
+        case .layers:
+            return "Layers"
+        case .bookmarks:
+            return "Bookmarks"
+        }
+    }
+}
+
+extension MapViewController {
+    func userRequestsExtras(_ barButtonItem: UIBarButtonItem?) {
+        guard mapViewMode != .disabled else {
+            return
+        }
+
+        let action = UIAlertController(title: nil, message: nil, preferredStyle: .actionSheet)
+                
+        // Present the list of extras.
+        for extra in Extras.allCases {
+            var extraAction: UIAlertAction
```

Comment:
This doesn't need to be mutable.

```suggestion
            let extraAction: UIAlertAction
```